pylint/checkers/utils.py

```diff
@@ -1522,3 +1522,19 @@ def get_iterating_dictionary_name(
         return node.iter.as_string()
 
     return None
+
+
+def get_subscript_const_value(node: astroid.Subscript) -> astroid.Const:
+    """
+    Returns the value (subscript.slice) of a Subscript node,
+    also supports python <3.9 windows where node.slice might be an Index
+    node
+    """
+    value = node.slice
+    if isinstance(value, astroid.Index):
+        value = value.value
+    inferred = safe_infer(value)
+    if not isinstance(inferred, astroid.Const):
+        raise ValueError("Subscript.slice cannot be inferred as an astroid.Const")
```

Comment:
Does it make sense to change that to a custom error instead of `ValueError`?

Reply:
I'll change this to ``astroid.InferenceError``

Reply:
on second thought, a custom error would be better, let me think of what to name it

Reply:
Wrote custom error: ``InferredTypeError``

Reply:
resolved : cd86ca0